hypertrace-graphql-gateway-service-metric-utils/src/main/java/org/hypertrace/graphql/utils/metrics/gateway/MetricAggregationContainerMapConverter.java

```diff
@@ -39,15 +41,15 @@
         .collect(immutableMapEntryCollector());
   }
 
-  private Single<Entry<String, MetricAggregationContainer>> buildMetricContainerEntry(
+  private Single<Entry<String, BaselineMetricAggregationContainer>> buildMetricContainerEntry(
       GroupedObservable<AttributeModel, MetricAggregationRequest> requestsForAttribute,
       Map<String, AggregatedMetricValue> metricResponses) {
 
     return requestsForAttribute
         .toList()
         .flatMap(
-            metricRequests -> this.aggregationMapConverter.convert(metricRequests, metricResponses))
-        .map(ConvertedAggregationContainer::new)
+            metricRequests -> this.aggregationMapConverter.convert(metricRequests, metricResponses, Collections.EMPTY_MAP))
+        .map(aggMap -> new BaselineConvertedAggregationContainer((Map<MetricLookupMapKey, BaselinedMetricAggregation>) aggMap))
```

Comment:
Why is the cast needed? can we fix the types and remove the cast, then switch to `BaselineConvertedAggregationContainer::new`